frontend/Index.svelte

```diff
@@ -24,243 +26,119 @@
-        s: "ص",
-        d: "ڈ",
-        f: "أ",
-        g: "غ",
-        h: "ھ",
-        j: "ض",
-        k: "خ",
-        z: "ذ",
-        x: "ژ",
-        c: "ث",
-        v: "ظ",
-        n: "ں",
-      },
-      notShift: {
-        "1": "۱",
-        "2": "۲",
-        "3": "۳",
-        "4": "۴",
-        "5": "۵",
-        "6": "٦",
-        "7": "۷",
-        "8": "۸",
-        "9": "۹",
-        "0": "۰",
-        q: "ق",
-        w: "و",
-        e: "ع",
-        r: "ر",
-        t: "ت",
-        y: "ے",
-        u: "ء",
-        i: "ی",
-        o: "ہ",
-        p: "پ",
-        a: "ا",
-        s: "س",
-        d: "د",
-        f: "ف",
-        g: "گ",
-        h: "ح",
-        j: "ج",
-        k: "ک",
-        l: "ل",
-        z: "ز",
-        x: "ش",
-        c: "چ",
-        v: "ط",
-        b: "ب",
-        n: "ن",
-        m: "م",
-        ",": "،",
-        ".": "۔",
-        "?": "؟",
-        ";": "؛",
-      },
-    },
-    hindi: {
-      shift: {},
-      notShift: {
-        "1": "१",
-        "2": "२",
-        "3": "३",
-        "4": "४",
-        "5": "५",
-        "6": "६",
-        "7": "७",
-        "8": "८",
-        "9": "९",
-        "0": "०",
-      },
-    },
-  };
 </script>
 
-<Block {visible} {elem_id} {elem_classes} {scale} {min_width} allow_overflow={false} padding={true}>
+<Block
+  {visible}
+  {elem_id}
+  {elem_classes}
+  {scale}
+  {min_width}
+  allow_overflow={false}
+  padding={true}
+>
   {#if loading_status}
-    <StatusTracker autoscroll={gradio.autoscroll} i18n={gradio.i18n} {...loading_status} />
+    <StatusTracker
+      autoscroll={gradio.autoscroll}
+      i18n={gradio.i18n}
+      {...loading_status}
+    />
   {/if}
 
-  <label class:container>
+  <label class="container">
     <BlockTitle {show_label} info={undefined}>{label}</BlockTitle>
 
     <input
       data-testid="textbox"
       type="text"
       class="scroll-hide"
       bind:value
-      bind:this={textAreaEl}
+      bind:this={keyboard.targetElement}
       {placeholder}
       disabled={!interactive}
-      dir={isLanguageRTL(currentLanguage) ? "rtl" : "ltr"}
-      on:keypress={handle_keypress}
+      dir={keyboard.isCurLanguageRTL() ? "rtl" : "ltr"}
     />
-    <p class="transliteration" bind:this={transliteration}></p>
+    <p class="transliteration" bind:this={keyboard.transliteration}></p>
   </label>
 
   <div class="dropdown">
+    <!-- <button on:click={keyboard.toggle.bind(keyboard)} class="expand-button"> -->
     <button on:click={toggle} class="expand-button">
       <span class="expand-symbol">⌨</span>
       <span class="description">On-Screen Keyboard</span>
     </button>
     <hr class="divider" />
-    <select id="language" class="language-selector" bind:value={currentLanguage}>
-      {#each languages as language}
-        <option value={language.toLowerCase()}>{language}</option>
+    <select
+      id="language"
+      class="language-selector"
+      bind:value={keyboard.selectedLanguage}
```

Comment:
lets expose the selected language to python as well so the user can link the selected language with other elements or update the keyboard language based on the other element.